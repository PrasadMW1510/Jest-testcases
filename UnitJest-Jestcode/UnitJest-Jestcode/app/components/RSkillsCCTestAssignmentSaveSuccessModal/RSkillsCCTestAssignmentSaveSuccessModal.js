/**
 *
 * RSkillsCCTestAssignmentSaveSuccessModal
 *
 */
import PropTypes from 'prop-types';
import React from 'react';

import SAMButton from 'components/SAMButton';
import SAMLinkButton from 'components/SAMLinkButton';
import SAMModal from 'components/SAMModal';
import './RSkillsCCTestAssignmentSaveSuccessModal.scss';

function RSkillsCCTestAssignmentSaveSuccessModal({
  affectedMessage,
  isOpen,
  onYes,
  redirectToRoster,
  rSkillsTestName,
  title,
  unaffectedUsers,
  unenrolledUsers,
}) {
  function unaffectedUsersMsg() {
    if (unaffectedUsers.length > 0) {
      return (
        <div className="rskillscc-test-assignment-save-success__unaffected-users-message">
          The following students have already been assigned {rSkillsTestName}:
          {renderUnaffectedUnenrolledUsers(unaffectedUsers)}
        </div>
      );
    }
    return null;
  }

  function renderUnaffectedUnenrolledUsers(userArray) {
    const unaffectedUsersLi = userArray.map(u => (
      <li key={u.user_id}>
        {u.first_name} {u.last_name}
      </li>
    ));
    return (
      <ul className="rskillscc-test-assignment-modal-save-success__list">{unaffectedUsersLi}</ul>
    );
  }

  function unenrolledUsersMsg() {
    if (unenrolledUsers.length > 0) {
      return (
        <div className="rskillscc-test-assignment-save-success__unenrolled-users-message">
          The following students are not enrolled in rSkills Tests:
          {renderUnaffectedUnenrolledUsers(unenrolledUsers)}
        </div>
      );
    }
    return null;
  }

  function renderCloseButton() {
    const cssClass = 'rskillscc-test-assignment-save-success__ok-button';
    if (redirectToRoster) {
      return (
        <SAMLinkButton
          to="/roster"
          id="ok"
          buttonClassModifier={cssClass}
          onClickHandler={onYes}
          isPrimaryButton
        >
          OK
        </SAMLinkButton>
      );
    }
    return (
      <SAMButton buttonClassModifier={cssClass} onClickHandler={onYes} isPrimaryButton>
        OK
      </SAMButton>
    );
  }

  return (
    <SAMModal
      isOpen={isOpen}
      contentLabel=""
      modalClassModifier="rskillscc-test-assignment-modal-save-success"
    >
      <div className="rskillscc-test-assignment-save-success__container">
        <div className="rskillscc-test-assignment-save-success__title">{title}</div>
        <div className="rskillscc-test-assignment-save-success__info-container">
          <div className="rskillscc-test-assignment-save-success__info">
            <div className="rskillscc-test-assignment-save-success__test-name">
              {rSkillsTestName}
            </div>
            <div className="rskillscc-test-assignment-save-success__message">{affectedMessage}</div>
            {unaffectedUsersMsg()}
            {unenrolledUsersMsg()}
          </div>
        </div>
      </div>
      <div className="rskillscc-test-assignment-save-success__ok-button-container">
        {renderCloseButton()}
      </div>
    </SAMModal>
  );
}

RSkillsCCTestAssignmentSaveSuccessModal.defaultProps = {
  redirectToRoster: false,
  unaffectedUsers: [],
  unenrolledUsers: [],
};

RSkillsCCTestAssignmentSaveSuccessModal.propTypes = {
  affectedMessage: PropTypes.string.isRequired,
  isOpen: PropTypes.bool,
  onYes: PropTypes.func.isRequired,
  redirectToRoster: PropTypes.bool,
  rSkillsTestName: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  unaffectedUsers: PropTypes.array,
  unenrolledUsers: PropTypes.array,
};

export default RSkillsCCTestAssignmentSaveSuccessModal;
