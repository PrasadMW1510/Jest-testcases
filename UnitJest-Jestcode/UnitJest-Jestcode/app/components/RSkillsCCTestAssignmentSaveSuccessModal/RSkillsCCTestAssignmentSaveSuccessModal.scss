.rskillscc-test-assignment-modal-save-success {
  height: 250px;
  width: 320px;
  padding: 7px 5px 10px;
  overflow: hidden;
}

.rskillscc-test-assignment-save-success__container {
  border: 1px solid #000;
  min-height: 175px;
  overflow: auto;
  width: 300px;
}

.rskillscc-test-assignment-save-success__title {
  vertical-align: center;
  text-align: center;
  align-items: center;
  align-content: center;
  background-color: #ffa200;
  color: #fff;
  height: 25px;
  margin-top: 0;
  z-index: +1;
}

.rskillscc-test-assignment-save-success__test-name {
  font-weight: normal;
  font-size: 11px;
  margin-top: 10px;
  margin-bottom: 5px;
}

.rskillscc-test-assignment-save-success__message {
  font-weight: normal;
  font-size: 11px;
  height: 50px;
  margin-bottom: 5px;
}

.rskillscc-test-assignment-save-success__unaffected-users-message {
  font-weight: normal;
  font-size: 11px;
}

.rskillscc-test-assignment-save-success__unenrolled-users-message {
  font-weight: normal;
  font-size: 11px;
}

.rskillscc-test-assignment-save-success__ok-button-container {
  width: 100%;
  align-content: center;
}

.rskillscc-test-assignment-save-success__ok-button {
  background-color: #ffa200;
  margin: 10px auto;
  display: block;
}

.rskillscc-test-assignment-modal-save-success__list {
  list-style-type: none;
  padding: 0;
  margin-top: 0;
}

.rskillscc-test-assignment-save-success__info-container {
  overflow: auto;
  height: 165px;
}

.rskillscc-test-assignment-save-success__info {
  margin: 0 10px 0 20px;
}
