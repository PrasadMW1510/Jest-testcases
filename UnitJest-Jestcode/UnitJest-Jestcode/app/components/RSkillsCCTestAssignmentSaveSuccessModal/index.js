/**
 *
 * RSkillsCCTestAssignmentSaveSuccessModal
 *
 */

export { default } from './RSkillsCCTestAssignmentSaveSuccessModal';
