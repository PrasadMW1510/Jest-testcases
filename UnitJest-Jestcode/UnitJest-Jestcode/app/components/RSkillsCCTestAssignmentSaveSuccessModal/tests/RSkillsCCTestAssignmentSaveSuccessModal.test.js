import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import RSkillsCCTestAssignmentSaveSuccessModal from '../index';

describe('<RSkillsCCTestAssignmentSaveSuccessModal />', () => {
  let wrapper = null;
  const mockOnYes = jest.fn();
  const mockMessage = 'Placeholder Message';
  const mockRSkillsTestName = 'Workshop 1';
  const mockTitle = 'rBook Flex';

  it('should render correctly', () => {
    wrapper = shallow(
      <RSkillsCCTestAssignmentSaveSuccessModal
        affectedMessage={mockMessage}
        onYes={mockOnYes}
        rSkillsTestName={mockRSkillsTestName}
        title={mockTitle}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  describe('when redirectToRoster is true', () => {
    const mockUnaffectedUsers = [{ user_id: '1', first_name: 'John', last_name: 'Doe' }];
    const mockUnenrolledUsers = [{ user_id: '2', first_name: 'Jane', last_name: 'Doe' }];
    it('should render correctly', () => {
      wrapper = shallow(
        <RSkillsCCTestAssignmentSaveSuccessModal
          affectedMessage={mockMessage}
          onYes={mockOnYes}
          redirectToRoster
          rSkillsTestName={mockRSkillsTestName}
          title={mockTitle}
          unaffectedUsers={mockUnaffectedUsers}
          unenrolledUsers={mockUnenrolledUsers}
        />
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
