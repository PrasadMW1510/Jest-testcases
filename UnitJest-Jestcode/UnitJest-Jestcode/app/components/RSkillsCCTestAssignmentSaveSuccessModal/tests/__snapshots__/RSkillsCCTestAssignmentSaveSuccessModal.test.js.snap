// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RSkillsCCTestAssignmentSaveSuccessModal /> should render correctly 1`] = `
<SAMModal
  contentLabel=""
  modalClassModifier="rskillscc-test-assignment-modal-save-success"
>
  <div
    className="rskillscc-test-assignment-save-success__container"
  >
    <div
      className="rskillscc-test-assignment-save-success__title"
    >
      rBook Flex
    </div>
    <div
      className="rskillscc-test-assignment-save-success__info-container"
    >
      <div
        className="rskillscc-test-assignment-save-success__info"
      >
        <div
          className="rskillscc-test-assignment-save-success__test-name"
        >
          Workshop 1
        </div>
        <div
          className="rskillscc-test-assignment-save-success__message"
        >
          Placeholder Message
        </div>
      </div>
    </div>
  </div>
  <div
    className="rskillscc-test-assignment-save-success__ok-button-container"
  >
    <SAMButton
      buttonClassModifier="rskillscc-test-assignment-save-success__ok-button"
      buttonType="button"
      isPrimaryButton={true}
      onClickHandler={[Function]}
    >
      OK
    </SAMButton>
  </div>
</SAMModal>
`;

exports[`<RSkillsCCTestAssignmentSaveSuccessModal /> when redirectToRoster is true should render correctly 1`] = `
<SAMModal
  contentLabel=""
  modalClassModifier="rskillscc-test-assignment-modal-save-success"
>
  <div
    className="rskillscc-test-assignment-save-success__container"
  >
    <div
      className="rskillscc-test-assignment-save-success__title"
    >
      rBook Flex
    </div>
    <div
      className="rskillscc-test-assignment-save-success__info-container"
    >
      <div
        className="rskillscc-test-assignment-save-success__info"
      >
        <div
          className="rskillscc-test-assignment-save-success__test-name"
        >
          Workshop 1
        </div>
        <div
          className="rskillscc-test-assignment-save-success__message"
        >
          Placeholder Message
        </div>
        <div
          className="rskillscc-test-assignment-save-success__unaffected-users-message"
        >
          The following students have already been assigned 
          Workshop 1
          :
          <ul
            className="rskillscc-test-assignment-modal-save-success__list"
          >
            <li
              key="1"
            >
              John
               
              Doe
            </li>
          </ul>
        </div>
        <div
          className="rskillscc-test-assignment-save-success__unenrolled-users-message"
        >
          The following students are not enrolled in rSkills Tests:
          <ul
            className="rskillscc-test-assignment-modal-save-success__list"
          >
            <li
              key="2"
            >
              Jane
               
              Doe
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    className="rskillscc-test-assignment-save-success__ok-button-container"
  >
    <withRouter(SAMLinkButton)
      buttonClassModifier="rskillscc-test-assignment-save-success__ok-button"
      id="ok"
      isPrimaryButton={true}
      onClickHandler={[Function]}
      to="/roster"
    >
      OK
    </withRouter(SAMLinkButton)>
  </div>
</SAMModal>
`;
